### DISCUSSION TITLE
Using PyCharm with TM1Py

### DISCUSSION THREAD

**Discussion by TM1si** [cubewise-code/tm1py#1015] [General]
When searching for a good editor for developing with TM1Py I found PyCharm. It works nice with the tutorial examples, but using a simple TM1Py script, which runs perfectly in terminal/console I get an errror:
```

File "D:\PowerPlanT\TM1\TM1PY\PycharmProjects\pythonProject\main.py", line 12, in <module>
    from TM1py import TM1Service, Subset
ModuleNotFoundError: No module named 'TM1py'
```

Any idea, where I have to add code getting TM1Py recognized by PyCharm?

**Comment by MariusWirtz** [comment]
The problem might be that PyCharm by default suggests using python with a virtual environment.

If you configure the project to use the system interpreter, you can avoid virtual environments.

<img width="1049" alt="image" src="https://github.com/cubewise-code/tm1py/assets/15182131/b84da0a1-b347-4671-8e09-b2aa7d24403e">

**Comment by TM1si** [comment]
Thanks a lot. This works fine with my TM1-Code.
